### PR TITLE
WIP:  Openstack: adds openstack ovsdpdk workflow

### PR DESCRIPTION
**Pull request by adduarte**

### CODE REVIEW

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ref:
+  as: openstack-ovsdpdk-bindvfiopcidriver
+  from: openstack-installer
+  commands: openstack-ovsdpdk-bindvfiopcidriver-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  env:
+    - name: VFIONETWORK
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml:10]: you'll need OS_CLOUD

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh:6]: you'll need to load the clouds.yaml here:

```
export OS_CLOUD
export OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"
```

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        contents:
+          source: data:;base64,b3B0aW9ucyB2ZmlvIGVuYWJsZV91bnNhZmVfbm9pb21tdV9tb2RlPTEK
+EOF
+echo "Creating VFIO NoIOMMU MachineConfig"
+oc create -f 05-vfio-noiommu.yaml
+
+git clone https://github.com/rh-nfv-int/shift-on-stack-vhostuser/
+cd shift-on-stack-vhostuser
+NETWORK_ID=$(openstack network show ${VFIONETWORK} -f value -c id)
+ansible-playbook play.yaml -e network_ids="$NETWORK_ID"
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh:33]: I don't know if ansible is installed (I suspect not), you'll need to install that in the container image.

Comment (EmilienM) [reply ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh:33]: I don't know if we need to load KUBECONFIG or if it's done automagically alraedy.

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+ref:
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml:1]: you need OWNERS and metadata file

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh (hunk 3 above)
Comment (EmilienM) [reply ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh:33]: At least I would add:


```
if test ! -f "${KUBECONFIG}"
then
	echo "No kubeconfig, can not continue."
	exit 0
fi
```

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh (hunk 2 above)
Comment (adduarte) [reply ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh:6]: instead we'll just take the network id as input to the step. Either directly or from shared_dir. We have to create the network anywahys. and using the name to look it up is a wasted cycle.

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml (hunk 1 above)
Comment (adduarte) [reply ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml:10]: We should not use any openstack commands here. We just need the network id. which we should already have.

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh (hunk 3 above)
Comment (adduarte) [reply ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-machineset-commands.sh:33]: ansible is installed on the openstack-installer image which is what we'll use to execute this. Now the oc command might be a problem, but in reality we should only craste the yaml files and drop them where required to be used by another step. 
but for now lets use the oc in the openstack-installer just to get this going.  
And it looks like we have to set K8S_AUTH_KUBECONFIG

to the pathc of kubeconfig. 
That is available to the step as ${KUBECONFIG}

File: ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml (hunk 4 above)
Comment (adduarte) [reply ci-operator/step-registry/openstack/ovsdpdk/machineset/openstack-ovsdpdk-bindvfiodriver-ref.yaml:1]: done

**Comment by adduarte** [comment]
Last push made drastic changes. 
Renamed step to ovsdpdk-workerms, as this step should really result in the creation of a machineset  which maybe added to a worker to connect it to an vfio network (ovs-dpdk, vhostuser, etc...)

**Comment by EmilienM** [comment]
https://github.com/openshift/release/pull/23828

**Comment by adduarte** [comment]
#23828

**Pull request closed without merging**